### PR TITLE
Enhanced RDoc for Struct

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #values_at
- #select
- #==
- #hash
- #eql?
- #size
- #dig

### CODE REVIEW

File: struct.c
Comment (jeremyevans) [inline struct.c:1492]: I think `number` is better than `count` here.  `count` should be used when something requires an action per thing being counted, whereas the size of a struct is always fixed per class.

File: struct.c
Comment (jeremyevans) [inline struct.c:1322]: This explanation is too long winded and goes into too much implementation detail. Maybe instead: `With a block given, returns an array of values from +self+ for which the block returns a truthy value`

File: struct.c
Comment (jeremyevans) [inline struct.c:1294]: You can remove the TypeError line, as explained in the documentation guide section about exceptions.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for working on this.  Only a few requested changes.

File: struct.c
Comment (BurdetteLamar) [reply struct.c]: Done (I think).

File: struct.c
Comment (BurdetteLamar) [reply struct.c]: Done.

File: struct.c
Comment (BurdetteLamar) [reply struct.c]: Done.

File: struct.c
Comment (BurdetteLamar) [reply struct.c]: Done.

**Review by jeremyevans — APPROVED** [review]
Looks good.  Please wait until CI passes before merging this, as a non-comment brace was removed and then added back.

**Pull request merged**
